File: backend/src/main/java/com/java/NBE4_5_1_7/domain/study/entity/StudyContent.java
package com.java.NBE4_5_1_7.domain.study.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class StudyContent {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long study_content_id;

    @Column(name = "first_category")
    @Enumerated(EnumType.STRING)
    private FirstCategory firstCategory;

    @Column(name = "second_category")
    private String secondCategory;

    private String title;

    @Lob
    @Column(name = "body", columnDefinition = "TEXT")
    private String body;
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/study/entity/StudyContent.kt
package com.java.NBE4_5_3_7.domain.study.entity

import jakarta.persistence.*

@Entity
class StudyContent() {
    @JvmField
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var study_content_id: Long? = null

    @JvmField
    @Column(name = "first_category")
    @Enumerated(EnumType.STRING)
    var firstCategory: FirstCategory? = null

    @JvmField
    @Column(name = "second_category")
    var secondCategory: String? = null

    @JvmField
    var title: String? = null

    @JvmField
    @Lob
    @Column(name = "body", columnDefinition = "TEXT")
    var body: String? = null

    constructor(title: String, body: String, firstCategory: FirstCategory, secondCategory: String) : this() {
        this.title = title
        this.body = body
        this.firstCategory = firstCategory
        this.secondCategory = secondCategory
    }
}
